### PR TITLE
Clean Up API Nodes

### PR DESCRIPTION
**Pull request by noanflaherty**
Does everything needed to make sure that when you pull down an `APINode`, it generates valid code.

### CODE REVIEW

File: ee/codegen/src/__test__/nodes/__snapshots__/api-node.test.ts.snap [hunk 1]
```diff
@@ -58,10 +58,10 @@ class APINodeDisplay(BaseAPINodeDisplay[APINode]):
 `;
 
 exports[`ApiNode > basic > getNodeFile 1`] = `
-"from vellum.workflows.nodes.displayable import ApiNode
+"from vellum.workflows.nodes.displayable import APINode as BaseAPINode
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/nodes/__snapshots__/api-node.test.ts.snap:61]: The actual base node had been renamed.

File: src/vellum/workflows/nodes/displayable/bases/api_node/node.py [hunk 2]
```diff
@@ -26,11 +26,11 @@ class BaseAPINode(BaseNode, Generic[StateType]):
     url: str
     method: APIRequestMethod
     data: Optional[str] = None
-    json: Optional["JsonObject"] = None
+    json: Optional["Json"] = None
```
Comment (noanflaherty) [inline src/vellum/workflows/nodes/displayable/bases/api_node/node.py:29]: Cleans up some types to what I claim to be more correct.

File: src/vellum/workflows/utils/vellum_variables.py [hunk 3]
```diff
@@ -32,6 +33,17 @@ def primitive_type_to_vellum_variable_type(type_: Union[Type, BaseDescriptor]) -
...
+            # Matches the type found at vellum.workflows.utils.vellum_variables.Json
+            if types == [
+                bool,
+                int,
+                float,
+                str,
+                typing.List[typing.ForwardRef('Json')],  # type: ignore [misc]
+                typing.Dict[str, typing.ForwardRef('Json')],  # type: ignore [misc]
+            ]:
+                return "JSON"
```
Comment (noanflaherty) [inline src/vellum/workflows/utils/vellum_variables.py:46]: This is kinda funky, but it works. Without this, a test fails.

**Comment by noanflaherty** [comment]
Open to post-push feedback on this! Merging so I can test e2e

File: src/vellum/workflows/utils/vellum_variables.py (hunk 3 above)
Comment (dvargasfuertes) [reply src/vellum/workflows/utils/vellum_variables.py:46]: [oos] wondering if we should do a set comparison (and if `BaseDescriptor.types` should actually be a `set`...)

**Review by dvargas92495 — COMMENTED** [review]
✅

**Pull request merged**